components/TranscriptFilter.tsx

```diff
@@ -0,0 +1,80 @@
+import { MenuItem, Select } from '@mui/material';
+import { DatePicker } from "@mui/x-date-pickers";
+import { isAfter, isBefore } from "date-fns";
+
+export type DateRange = {
+  start: Date | null,
```

Comment:
What about using 0 and something way off in the future for unset?  Then the type is simpler..

Reply:
I would tend to disagree. The type would be hidden, not simpler. If I can safely assume that any non-null date object is valid, then it prevents having to check at display/conversion time.